mm2src/coins/nft.rs

```diff
@@ -753,3 +804,59 @@ where
     }
     Ok(())
 }
+
+fn check_and_redact_if_spam(value: &mut Option<String>) -> bool {
+    match value {
+        Some(s) if s.contains("http://") || s.contains("https://") => {
+            *value = Some("URL redacted for user protection".to_string());
+            true
+        },
+        _ => false,
+    }
+}
+
+fn protect_from_history_spam(tx: &mut NftTransferHistory) {
+    let collection_name_spam = check_and_redact_if_spam(&mut tx.collection_name);
+    let token_name_spam = check_and_redact_if_spam(&mut tx.token_name);
+
+    if collection_name_spam || token_name_spam {
+        tx.possible_spam = Some(true);
```

Comment:
Why does `possible_spam` has to be an `option`? Can't it just be a non-optional bool.

Reply:
`possible_spam` is a field from moralis response. I didnt see `possible_spam` filed not null, but decided to protect us from this situation, bcz it is the new feature, and they still didnt provide info which fileds can be optional.

Well, as Tony noticed that sometimes they mark [obviously spam NFT as not spam](https://github.com/KomodoPlatform/komodowallet/issues/1128#issuecomment-1592939000), I believe we can have `possible_spam` false as default.

Reply:
Done, in Wrapper structures for Nft and History I set `#[serde(default)]` annotation for `possible_spam`